cartographer/mapping/trajectory_builder.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_TRAJECTORY_BUILDER_H_
+#define CARTOGRAPHER_MAPPING_TRAJECTORY_BUILDER_H_
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "cartographer/common/make_unique.h"
+#include "cartographer/common/port.h"
+#include "cartographer/common/time.h"
+#include "cartographer/kalman_filter/pose_tracker.h"
+#include "cartographer/mapping/submaps.h"
+#include "cartographer/sensor/data.h"
+#include "cartographer/sensor/laser.h"
+#include "cartographer/sensor/point_cloud.h"
+
+namespace cartographer {
+namespace mapping {
+
+// This interface is used for both 2D and 3D SLAM.
+class TrajectoryBuilder {
+ public:
+  // Represents a newly computed pose. Each of the following steps in the pose
+  // estimation pipeline are provided for debugging:
+  //
+  // 1. UKF prediction
+  // 2. Absolute pose observation (e.g. from scan matching)
+  // 3. UKF estimate after integrating any measurements
+  //
+  // Finally, 'pose' is the end-user visualization of orientation and
+  // 'point_cloud' is the point cloud, in the local map frame.
+  struct PoseEstimate {
+    PoseEstimate() = default;
+    PoseEstimate(common::Time time,
```

Comment:
change parameters to start with init_ to avoid referring to two different things by the same name